tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/InstanceDiscoveryManagerTests.cs

```diff
@@ -119,6 +122,80 @@ public async Task KnownMetadataProviderIsCheckedSecondAsync()
             _knownMetadataProvider.Received(1).GetMetadata("some_env.com", otherEnvs);
         }
 
+        [TestMethod]
+        public async Task AuthorityValidationFailure_IsRethrown_Async()
+        {
+            // Arrange
+            var validationException = new MsalServiceException(MsalError.InvalidInstance, "authority validation failed"); 
+            _staticMetadataProvider = new StaticMetadataProvider();
+
+            // network fails with invalid_instance exception
+            _networkMetadataProvider
+                .When(x => x.FetchAllDiscoveryMetadataAsync(Arg.Any<Uri>(), _testRequestContext))
+                .Do(x => throw validationException);
+          
+
+            // Act
+            var actualException = await AssertException.TaskThrowsAsync<MsalServiceException>(() =>
+            _discoveryManager.GetMetadataEntryAsync(
+                "https://some_env.com/tid",
+                _testRequestContext))
+                .ConfigureAwait(false);
+
+            // Assert
+            Assert.AreSame(validationException, actualException);
+            _knownMetadataProvider.DidNotReceiveWithAnyArgs();
+        }
+
+        [TestMethod]
+        public async Task NetworkProviderFailures_AreIgnored_Async()
+        {
+            // Arrange
+            _staticMetadataProvider = new StaticMetadataProvider();
+            _knownMetadataProvider.GetMetadata("some_env.com", Enumerable.Empty<string>()).Returns(_expectedResult);
+
+            // network fails with something other than invalid_instance exception
+            _networkMetadataProvider
+                .When(x => x.FetchAllDiscoveryMetadataAsync(Arg.Any<Uri>(), _testRequestContext))
+                .Do(x => throw new MsalServiceException("endpoint_busy", "some exception message"));
+
+
+            // Act
+            var actualResult = await _discoveryManager.GetMetadataEntryAsync(
+                "https://some_env.com/tid",
+                _testRequestContext)
+                .ConfigureAwait(false);
+
+            // Assert
+            Assert.AreSame(_expectedResult, actualResult, "The known metadata provider should be queried second");
+            _knownMetadataProvider.Received(1).GetMetadata("some_env.com", Enumerable.Empty<string>());
+        }
+
+        [TestMethod]
+        public async Task NetworkProviderFailures_WithNoKnownMetadata_ContinuesWithAuthority_Async()
+        {
+            // Arrange
+            _staticMetadataProvider = new StaticMetadataProvider();
+
+            // no known metadata 
+            _knownMetadataProvider.GetMetadata(null, null).ReturnsForAnyArgs((InstanceDiscoveryMetadataEntry)null);
+
+            // network fails with something other than invalid_instance exception
+            _networkMetadataProvider
+                .When(x => x.FetchAllDiscoveryMetadataAsync(Arg.Any<Uri>(), _testRequestContext))
+                .Do(x => throw new MsalServiceException("endpoint_busy", "some exception message"));
```

Comment:
>endpoint_busy [](start = 57, length = 13)

is this an actual exception we can get?

Reply:
No, it's an exception that is different from "invalid_instance".